src/components/BookList.jsx

```diff
@@ -7,7 +7,7 @@ const BookList = props => {
   return(
     <div className='bookList'>
       {books.map(book => (
-        <BookItem title={book.title} author={book.author} key={book.id} id={book.id}/>
+        <BookItem title={book.title} author={book.author} key={book.item_id} id={book.item_id}/>
       ))}
     </div>
```

Comment:
- [ ] <b>REQUIRED:</b> Your component is still not listing the data. This is due to the fact that when the component renders, there is not yet data in the state. so how can you fix that?  simply make sure that when the component renders for the second time,  the data will be in the state. and to do that,  first import `useEffect` from react. 

```jsx
import React, { useEffect } from 'react';
```
We also gonna need dispatch, so we import useDispatch from react-redux. 
and getBookList action.

```jsx
import { useDispatch } from 'react-redux';
import { getBookList } from '../redux/books/books';
```
The finally
```jsx
const dispatch = useDispatch();
const books = useSelector((state) => state.books, shallowEqual);
useEffect(() => { dispatch(getBookList()); }, []);
```